### ISSUE TITLE
DeDRM Macintosh Application does not work

### ISSUE THREAD

**Issue by iseens** [apprenticeharper/DeDRM_tools#505]
I can't use the "DeDRM.app" for Mac to remove DRM now.

- I use "DeDRM.app" for Mac in "DeDRM tools 6.6.0".
- I use macOS High sierra version 10.13.4.
- I download ebook from "Kindle" for Mac 1.17.1 (KindleForMac-44182.dmg).
- [DeDRM log.txt](https://github.com/apprenticeharper/DeDRM_tools/files/1990491/DeDRM.log.txt)

Thanks.

**Comment by ElleKayEm** [comment]
Are you sure Kindle for Mac hasn't updated to a more recent version?  Because I'm wondering if this is in KFX format and whether or not the stand alone app can handle that.

If that's not it, then perhaps the key it's finding is actually from a previous installation of Kindle for Mac.

**Comment by j-howell** [comment]
From the log it looks to me like the import of ion.py failed. Unfortunately when that happens it assumes that it must be running as a calibre plugin and attempts to import the same module again, hiding the real cause of the error.

I suspect a missing dependency in the standalone python, perhaps lzma.

**Comment by fireattack** [comment]
It does look like the same issue (#512) I encountered, which is indeed caused by missing lzma

**Comment by isjamesalive** [comment]
@j-howell Yeah I think so! I got this working by installing backports.lzma and pycrypto with pip. Python2.7 on High Sierra doesn't seem to include the site-packages directory where pip installs them so I just patched ion.py:

```patch
--- ion.py	2018-05-25 12:50:10.000000000 +0930
+++ ion_fixed.py	2018-05-25 12:50:17.000000000 +0930
@@ -11,6 +11,9 @@
 import os
 import os.path
 import struct
+import sys
+
+sys.path.append('/usr/local/lib/python2.7/site-packages/')

 try:
     from cStringIO import StringIO
```

**Comment by apprenticeharper** [comment]
I'm hoping that this issue is fixed with 6.6.1, but I haven't been able to check.

**Comment by cdkii** [comment]
This fixed mine.

`chmod +x DeDRM.app/Contents/MacOS/*`

You need the folder for the location of the Mac app, so if its in your downloads folder, it would be as follows:

`chmod +x /Users/**(your username)**/Downloads/DeDRM_tools_6/DeDRM_Macintosh_Application/DeDRM.app/Contents/MacOS/*`

Then run as normal.

**Comment by yosefrajwulf** [comment]
@cdkii Thanks! That also worked for me.

**Comment by apprenticeharper** [comment]
It seems that copying the files to create the archive wipes out the flag on Windows that zip interprets as executable on Unix systems. If anyone has a fix for this, please patch the copying script.

**Comment by JM0604** [comment]
I know this is an old thread but my DeDRM 6.6.3 won't open with Mac.  I am getting the same error as above.  I see someone posted how to fix it but I am not familiar with this.  How do you apply that fix to it so that it opens?

**Comment by ElleKayEm** [comment]
Take a look at the FAQs here:

https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#the-macintosh-application

**Comment by JM0604** [comment]
thank you! i think I need to use the command thing but I am not familiar with that.  I'll try and figure it out! Thank you!
